src/git-data.ts

```diff
@@ -82,24 +82,44 @@ export class GitData {
 
     private async initRemoteData (cwd: string, writeStreams: WriteStreams): Promise<void> {
         try {
-            const {stdout: gitRemote} = await Utils.spawn(["git", "remote", "-v"], cwd);
-            const gitRemoteMatch = /(?<schema>git|https?)(?::\/\/|@)(?<host>[^:/]*)(:(?<port>\d+)\/|:|\/)(?<group>.*)\/(?<project>[^ .]+)(?:\.git)?.*/.exec(gitRemote);
-
-            assert(gitRemoteMatch?.groups != null, "git remote -v didn't provide valid matches");
-
-            this.remote.schema = gitRemoteMatch.groups.schema;
-            if (this.remote.schema === "git") {
+            let gitRemoteMatch;
+            const {stdout: gitRemote} = await Utils.spawn(["bash", "-c", "git remote get-url gcl-origin 2> /dev/null || git remote get-url origin"], cwd);
```

Comment:
Can you explain the logic behind this?

Reply:
> Can you explain the logic behind this?
https://github.com/firecow/gitlab-ci-local/issues/921#issuecomment-2090224041
`gcl-origin` is to for people to customize their own "remote" in the event that `origin` is not what they want

Reply:
> If bash is really needed Utils.bash should be used.
> Keep in mind also, that bash is a lot slower, than just calling plain ol' shell cmd's

Ok, got it, wasnt aware of this.

Reply:
> Wouldn't this be cleaner?
LGTM

Reply:
> Can you explain the logic behind this?
> 
> ```ts
> let gitRemote;
> try {
>     const res = await Utils.spawn(["git", "remote", "get-url", "gcl-origin"], cwd);
>     gitRemote = res.stdout;
> } catch {
>     const res = await Utils.spawn(["git", "remote", "get-url", "origin"], cwd);
>     gitRemote = res.stdout;
> }
> ```
> 
> Wouldn't this be cleaner? (I don't like bash magic) 😄

yeah lgtm,though i am a bash magician 😂

Reply:
Haha, yeah... But it ramps up, spawning that many bash instances :smile: